File: app/src/main/java/com/mnj/java_solutions/stack/RemoveAllAdjacentDuplicates.java
package com.mnj.java_solutions.stack;

import java.util.Stack;

public class RemoveAllAdjacentDuplicates {

    public void removeDuplicate() {

        String str = "azxxzy";
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);
            if (!stack.isEmpty() && stack.peek() == ch) {
                stack.pop();
            } else {
                stack.push(ch);
            }
        }
        char[] result = new char[stack.size()];
        int index = stack.size()-1;
        while (!stack.isEmpty()) {
            result[index--] = stack.pop();
        }
        System.out.println("==>> Result: " + String.valueOf(result));
    }
}

File: app/src/main/java/com/mnj/java_solutions/stack/RemoveAllAdjacentDuplicates.kt
package com.mnj.java_solutions.stack

import java.util.*

class RemoveAllAdjacentDuplicates {
    fun removeDuplicate() {
        val str = "azxxzy"
        val stack = Stack<Char>()
        for (element in str) {
            if (!stack.isEmpty() && stack.peek() == element) {
                stack.pop()
            } else {
                stack.push(element)
            }
        }
        val result = CharArray(stack.size)
        var index = stack.size - 1
        while (!stack.isEmpty()) {
            result[index--] = stack.pop()
        }
        println("==>> Result: " + String(result))
    }
}